cypress/pageObject/Patients/PatientLocation.ts

```diff
@@ -0,0 +1,229 @@
+    if (status) {
+      cy.clickAndSelectOption(this.selectors.locationStatus, status);
+    }
+    return this;
+  }
+
+  selectOperationalStatus(status: string) {
+    if (status) {
+      cy.clickAndSelectOption(this.selectors.operationalStatus, status);
+    }
+    return this;
+  }
+
+  fillLocationData(data: LocationData) {
+    this.selectLocationForm(data.form)
+      .enterLocationName(data.name)
+      .enableBulkCreation(data.form === "Bed")
+      .selectBedCounts(data.bedsCount)
+      .enterDescription(data.description)
+      .selectLocationStatus(data.status)
+      .selectOperationalStatus(data.opStatus);
+    return this;
+  }
+
+  interceptLocationCreationRequest() {
+    cy.intercept("POST", `**/api/v1/facility/**`).as("createLocation");
+    return this;
+  }
+
+  verifyLocationCreationAPICall() {
+    cy.wait("@createLocation").its("response.statusCode").should("eq", 200);
+    return this;
+  }
+
+  submitLocationForm() {
+    cy.clickSubmitButton("Create");
+    return this;
+  }
+
+  assertLocationCreationSuccess() {
+    cy.verifyNotification("Location Created");
+    return this;
+  }
+
+  assertMultipleBedsCreationSuccess(bedCount: string) {
+    cy.verifyNotification(`${bedCount} created successfully`);
+    return this;
+  }
+
+  navigateToEncounters() {
+    cy.get('[data-sidebar="content"]').contains("Encounters").click();
+    return this;
+  }
+
+  clickPlannedEncounterFilter() {
+    cy.intercept("GET", "**/api/v1/encounter/**").as("getEncounters");
+    cy.verifyAndClickElement('[data-cy="planned-filter"]', "Planned");
+    cy.wait("@getEncounters", { timeout: 10000 }).then((interception) => {
+      expect(interception.request.url).to.include("status=planned");
+      expect(interception.response.statusCode).to.eq(200);
+    });
+    return this;
+  }
+
+  openFirstEncounterDetails() {
+    cy.get('[data-cy="encounter-list-cards"]')
+      .first()
+      .contains("View Details")
+      .click();
+    return this;
+  }
+
+  clickAssociatedLocationBadge() {
+    cy.get(this.selectors.currentLocationBadge).click();
+    return this;
+  }
+
+  clickAddLocationBadge() {
+    cy.get(this.selectors.addLocationBadge).click();
+    return this;
+  }
+
+  clickUpdateLocationButton() {
+    cy.verifyAndClickElement(
+      this.selectors.updateLocationButton,
+      "Update Location",
+    );
+    return this;
+  }
+
+  searchBedLocation(locationName: string) {
+    cy.typeAndSelectOption(this.selectors.locationSearchTrigger, locationName);
+    return this;
+  }
+
+  submitLocationAssociation() {
+    cy.verifyAndClickElement(
+      this.selectors.submitLocationAssociation,
+      "Create Location Association",
+    );
+    return this;
+  }
+
+  setStatusCompleted() {
+    cy.clickAndSelectOption(
+      this.selectors.associatedLocationStatus,
+      "Completed",
+    );
+    return this;
+  }
+
+  fillEndTime(dateTime: string) {
+    cy.get('[data-cy="location-end-date-time"]').clear().type(dateTime);
```

Comment:
```suggestion
    cy.get('[data-cy="location-end-date-time"]').clear().type(dateTime);
```

cy.typeintofield command function for these kind of usages

Reply:
done